src/liballoc/string.rs

```diff
@@ -1732,18 +1732,31 @@ impl<'a> FromIterator<&'a str> for String {
 #[stable(feature = "extend_string", since = "1.4.0")]
 impl FromIterator<String> for String {
     fn from_iter<I: IntoIterator<Item = String>>(iter: I) -> String {
-        let mut buf = String::new();
-        buf.extend(iter);
-        buf
+        let mut iterator = iter.into_iter();
```

Comment:
Could you add a quick comment to this impl and the one below mentioning why they're structured this way, since it looks a bit weird at first glance.